src/main/java/com/redhat/devtools/intellij/qute/psi/internal/extensions/renarde/RenardeImplicitUsageProvider.java

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package com.redhat.devtools.intellij.qute.psi.internal.extensions.renarde;
+
+import com.intellij.codeInsight.daemon.ImplicitUsageProvider;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtilCore;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiModifier;
+import com.redhat.microprofile.psi.internal.quarkus.renarde.java.RenardeUtils;
+import org.jetbrains.annotations.NotNull;
+
+public class RenardeImplicitUsageProvider implements ImplicitUsageProvider {
```

Comment:
Add Javadoc to explain the renarde rule like you did with JavaEE ... class

Reply:
done